tests/ui/multiple_bound_locations.rs

```diff
@@ -0,0 +1,59 @@
+#![warn(clippy::multiple_bound_locations)]
+
+fn ty<F: std::fmt::Debug>(a: F)
+//~^ ERROR: bound is defined in more than one place
+where
+    F: Sized,
+{
+}
+
+fn lifetime<'a, 'b: 'a, 'c>(a: &'b str, b: &'a str, c: &'c str)
+//~^ ERROR: bound is defined in more than one place
+where
+    'b: 'c,
+{
+}
+
+fn ty_pred<F: Sized>()
+//~^ ERROR: bound is defined in more than one place
+where
+    for<'a> F: Send + 'a,
+{
+}
+
+struct B;
+
+impl B {
+    fn ty<F: std::fmt::Debug>(a: F)
+    //~^ ERROR: bound is defined in more than one place
+    where
+        F: Sized,
+    {
+    }
+
+    fn lifetime<'a, 'b: 'a, 'c>(a: &'b str, b: &'a str, c: &'c str)
+    //~^ ERROR: bound is defined in more than one place
+    where
+        'b: 'c,
+    {
+    }
+
+    fn ty_pred<F: Sized>()
+    //~^ ERROR: bound is defined in more than one place
+    where
+        for<'a> F: Send + 'a,
+    {
+    }
+}
+
+struct C<F>(F);
+
+impl<F> C<F> {
```

Comment:
What if we had another bound on F here?

Also: Should that even lint? Perhaps it's useful to have a secondary bound for just one method of the impl?

Reply:
I have no opinion about this, hence why I didn't add checks on impl blocks. So up to you. :)